src/backend/input.rs

```diff
@@ -0,0 +1,31 @@
+use std::hash::Hash;
+
+#[derive(Hash, PartialEq, Eq, Clone, Copy)]
+pub enum InputDevice<T: PartialEq + Eq + Hash + Clone + Copy>
```

Comment:
> So basically, it would be up to the computer user to decide whether their two keyboard should be treated as a single one, or appear as two different logical seats.

Logically seats are free to create and destroy as the user wishes right? So this is more of an issue on how to make this easily configurable, I see.